### PR TITLE
Ispn 6349 Primitive Stream API Changes

### PR DESCRIPTION
**Pull request by wburns**
https://issues.jboss.org/browse/ISPN-6349

### CODE REVIEW

File: core/src/main/java/org/infinispan/CacheStream.java [hunk 1]
```diff
@@ -639,5 +664,17 @@
     *               of new values
     * @return the new stream
     */
-   LongStream flatMapToLong(SerializableFunction<? super R, ? extends LongStream> mapper);
+   LongCacheStream flatMapToLong(SerializableFunction<? super R, ? extends LongStream> mapper);
+
+   @Override
+   CacheStream<R> parallel();
```
Comment (wburns) [inline core/src/main/java/org/infinispan/CacheStream.java]: Need to add javadocs here

**Comment by wburns** [comment]
Should be ready to go now.

**Comment by wburns** [comment]
Javadocs added.

**Comment by danberindei** [comment]
Integrated, thanks Will!

**Pull request closed without merging**